vulnerabilities/pipelines/samba_importer.py

```diff
@@ -0,0 +1,335 @@
+        return 0 if not security_table else len(security_table.find_all("tr")) - 1
+
+    def collect_advisories(self):
+        soup = BeautifulSoup(self.advisory_data, "html.parser")
+        security_table = soup.find("table", {"class": "security_table"})
+
+        if not security_table:
+            self.log("Security table not found in HTML content", level=logging.ERROR)
+            return
+
+        rows = security_table.find_all("tr")[1:]
+        for row in rows:
+            try:
+                advisory = self.parse_advisory_row(row)
+                if advisory:
+                    yield advisory
+            except Exception as e:
+                self.log(
+                    f"Error parsing advisory row: {e!r}:\n{traceback_format_exc()}",
+                    level=logging.ERROR,
+                )
+
+    def get_advisory_details(self, cve_id):
+        if cve_id in self.advisory_details_cache:
+            return self.advisory_details_cache[cve_id]
+
+        detail_url = f"https://www.samba.org/samba/security/{cve_id}.html"
+
+        try:
+            response = requests.get(detail_url)
+            response.raise_for_status()
+
+            soup = BeautifulSoup(response.text, "html.parser")
+            pre_tag = soup.find("pre")
+            if not pre_tag:
+                self.log(f"No detailed information found for {cve_id}", level=logging.WARNING)
+                return {}
+
+            announcement_text = html.unescape(pre_tag.get_text())
+            details = parse_announcement_text(announcement_text)
+            details["url"] = detail_url
+
+            self.advisory_details_cache[cve_id] = details
+            return details
+        except requests.exceptions.RequestException as e:
+            self.log(f"Error fetching advisory details for {cve_id}: {e}", level=logging.WARNING)
+            return {}
+        except Exception as e:
+            self.log(f"Error processing advisory details for {cve_id}: {e!r}", level=logging.ERROR)
+            return {}
+
+    def parse_advisory_row(self, row):
+        cells = row.find_all("td")
+        if len(cells) != 6:
+            self.log(
+                f"Expected 6 cells in security table row, got {len(cells)}", level=logging.WARNING
+            )
+            return None
+
+        date_issued = cells[0].get_text().strip()
+        patch_links = []
+
+        for link in cells[1].find_all("a"):
+            href = link.get("href", "")
+            if href:
+                if not href.startswith(("http://", "https://")):
+                    href = f"https://www.samba.org{href}"
+                patch_links.append(href)
+
+        issue_desc = cells[2].get_text().strip()
+        affected_releases = cells[3].get_text().strip()
+
+        cve_ids = []
+        for link in cells[4].find_all("a"):
+            cve_id = link.get_text().strip()
+            if re.match(r"CVE-\d{4}-\d{4,}", cve_id):
+                cve_ids.append(cve_id)
```

Comment:
CVE parsing is not clean it leads to `404 Client Error: Not Found for url: https://www.samba.org/samba/security/CVE-2019-19344..html
`